File: Example/Rudder-Sprig/_ViewController.m
//
//  _ViewController.m
//  Rudder-Sprig
//
//  Created by arnabp92 on 02/11/2020.
//  Copyright (c) 2020 arnabp92. All rights reserved.
//

#import "_ViewController.h"
#import <Rudder/Rudder.h>

@interface _ViewController ()

@end

@implementation _ViewController

RSClient *client;

- (void)viewDidLoad
{
    [super viewDidLoad];
	// Do any additional setup after loading the view, typically from a nib.

    client = [RSClient sharedInstance];
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)identifyAction:(UIButton *)sender {
    [client identify:@"Test_User_1" traits: @{@"email": @"test@rudderstack.com", @"firstName": @"First", @"lastName": @"Last"}];
}

- (IBAction)trackAction:(UIButton *)sender {
    [client track:@"Track Without Properties"];
}

- (IBAction)resetAction:(UIButton *)sender {
    [client reset:YES];
}

@end

File: Example/Rudder-Sprig/ViewController.swift
//
//  ViewController.swift
//  Rudder-Sprig_Example
//
//  Created by Abhishek Pandey on 16/10/24.
//  Copyright © 2024 arnabp92. All rights reserved.
//

import UIKit
import Rudder
import Rudder_Sprig

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
        
        // Pass ViewController instance to RudderSprigFactory. As Sprig SDK requires this to display the servey.
        RudderSprigFactory.instance.setViewController(self)
    }
    
    @IBAction func identify(_ sender: Any) {
        let traits: [String: Any] = [
            "email": "test@mail.com",
            "key-1": "value",
            "key-2": 100,
            "key-3": 122.56,
            "key-4": true,
        ]
        RSClient.getInstance().identify("User iOS 1", traits: traits)
    }
    
    @IBAction func trackWithoutProperties(_ sender: Any) {
        RSClient.getInstance().track("Track event without properties")
    }
    
    @IBAction func trackWithProperties(_ sender: Any) {
        let properties: [String: Any] = [
            "key-5": "value 5",
            "key-6": 200,
        ]
        RSClient.getInstance().track("Track event with properties", properties: properties)
    }
    
    @IBAction func reset(_ sender: Any) {
        RSClient.getInstance().reset(false)
    }
}
